src/script/sign.cpp

```diff
@@ -672,6 +685,7 @@ class DummySignatureChecker final : public BaseSignatureChecker
     bool CheckSchnorrSignature(Span<const unsigned char> sig, Span<const unsigned char> pubkey, SigVersion sigversion, ScriptExecutionData& execdata, ScriptError* serror) const override { return sig.size() != 0; }
     bool CheckLockTime(const CScriptNum& nLockTime) const override { return true; }
     bool CheckSequence(const CScriptNum& nSequence) const override { return true; }
+    bool CheckEqual(const std::vector<unsigned char>& vch1, const std::vector<unsigned char>& vch2) const override { return true; }
```

Comment:
In 8f1b45bf256b7454d1b790038a00eb537b9bbb70 "script/sign: signing support for Miniscripts with hash preimage challenges"

I think this line should be in the previous commit.

Reply:
Why? The previous commit does not need it, it only cares about the interpreter.

Reply:
The previous commit adds `CheckEqual`, whereas this commit adds `LookupPreimage`. It's not immediately obvious how these are related, and IMO it makes more sense to introduce the `CheckEqual`s in the same commit even if unneeded immediately.

Reply:
My rationale was that adding `CheckEqual` to the `DummySignatureChecker` is related to the descriptor: it makes a descriptor with timelocks reported as solvable (which in turn allows us to mark them as such in the unit tests). On the other hand the previous commit aims to be the isolated 3-lines-diff that touches consensus code, and only do that.

Not that i feel too strongly about it of course!